django/db/migrations/operations/models.py

```diff
@@ -744,25 +744,41 @@ def describe(self):
         return "Change managers on %s" % (self.name, )
 
 
-class AddIndex(Operation):
+class IndexOperation(Operation):
+
+    option_name = 'indexes'
+
+    @cached_property
+    def model_name_lower(self):
+        return self.model_name.lower()
+
+
+class AddIndex(IndexOperation):
     """
     Add an index on a model.
     """
 
     def __init__(self, model_name, index):
         self.model_name = model_name
+        try:
+            getattr(index, 'name')
+        except AttributeError:
+            raise ValueError("Index passed to AddIndex migration must have a name: %s doesn't" % index)
```

Comment:
"Indexes passed to AddIndex migration operations require a name attribute, %r doesn't."